src/lib/matcher.rs

```diff
@@ -407,3 +408,252 @@ impl MatcherFactory {
+        assert!(result.is_ok())
+    }
+    #[test]
+    fn test_validate_fixed_pattern_error() {
+        let pattern = "A.GTGTGATG";
+        let msg = String::from("Fixed pattern must contain only DNA bases: A .. [.] .. GTGTGATG");
+        let result = validate_fixed_pattern(&pattern);
+        let inner = result.unwrap_err().to_string();
+        assert_eq!(inner, msg);
+    }
+
+    // ############################################################################################
+    // Tests FixedStringMatcher::bases_match()
+    // ############################################################################################
+    #[rstest]
+    #[case(false, "GG", "AGGT", true)] // fixed pattern is found in bases, and invert_match is false
+    #[case(true, "GG", "AGGT", false)] // fixed pattern is found in bases, and invert_matches is true
+    #[case(true, "GG", "CCC", true)] // fixed pattern is not found in bases, and invert_match is true
+    #[case(false, "GG", "CCC", false)] // fixed pattern is bot found in bases, and invert_match is false
+    fn test_matcher_fixed_string_matcher_bases_match(
+        #[case] invert_match: bool,
+        #[case] pattern: &str,
+        #[case] bases: String,
+        #[case] expected: bool,
+    ) {
+        let opts = MatcherOpts {
+            invert_match,
+            reverse_complement: false, // does not impact FixedStringMatcher::bases_match()
+            color: false,
+        };
+
+        let matcher = FixedStringMatcher::new(pattern, opts);
+        assert_eq!(matcher.bases_match(bases.as_bytes()), expected);
+    }
+    // ############################################################################################
+    // Tests FixedStringSetMatcher::bases_match()
+    // ############################################################################################
+    #[rstest]
+    #[case(false, vec![("AGT".as_bytes().to_vec()), ("GG".as_bytes().to_vec())], "AGTGG", true)] // all patterns are found in bases, and invert_match is false
+    #[case(true, vec![("AGT".as_bytes().to_vec()), ("GG".as_bytes().to_vec())], "AGTGG", false)] // all patterns are found in bases, and invert_matches is true
+    #[case(false, vec![("AGT".as_bytes().to_vec()), ("TT".as_bytes().to_vec())], "AGTGG", true)] // some patterns are found in bases, and invert_matches is false
+    #[case(true, vec![("AGT".as_bytes().to_vec()), ("TT".as_bytes().to_vec())], "AGTGG", false)] // some patterns are found in bases, and invert_matches is true
+    #[case(false, vec![("CCC".as_bytes().to_vec()), ("TT".as_bytes().to_vec())], "AGTGG", false)] // no patterns are found in bases, and invert_matches is false
+    #[case(true, vec![("CCC".as_bytes().to_vec()), ("TT".as_bytes().to_vec())], "AGTGG", true)] // no patterns are found in bases, and invert_matches is true
+
+    fn test_matcher_fixed_string_set_matcher_bases_match(
+        #[case] invert_match: bool,
+        #[case] patterns: Vec<Vec<u8>>,
```

Comment:
why not have this be type `Vec<&str>` then do the `.as_bytes().to_vec()` within the function?